Review followup: PR #25620 — GH#25592: fix: ignore inactive queued check suites

## Unaddressed review bot suggestions

PR #25620 was merged with unaddressed review bot feedback. Each comment
below includes its file path, line number, a direct link to the inline
review comment, and a `diff` fence with the code context the bot was
flagging. Resolved and outdated threads are filtered out via GitHub's
GraphQL review-thread state. Read the relevant lines, decide whether
the suggestion is correct, and either apply the fix or close this issue
with a wontfix rationale.

**Source PR:** https://github.com/marcusquinn/aidevops/pull/25620

---

### You are the triager (worker-is-triager rule)

This issue is **auto-created from review bot output** and dispatched
directly to you. Review bots can be wrong: hallucinated line refs, false
premises about codebase structure, template-driven sweeps without
measurements (see GH#17832-17835 for prior art and `AGENTS.md`
"AI-Generated Issue Quality"). **Do not assume the bot is correct.** Verify before acting.

You must end in exactly one of three outcomes — no fourth "hand it back
to the human" path exists. Humans approve decisions; they do not re-do
analysis.

#### Outcome A — Premise falsified → close the issue

1. Read the cited `file:line` (listed under *Files to modify* below).
2. If the bot's claim is factually wrong (file doesn't exist at that
   line, function doesn't behave as described, "auto-generated" section
   isn't actually auto-generated, etc.), **close the issue** with a
   comment in this shape:

   > **Premise falsified.** \<what the bot claimed\>. \<what the code
   > actually shows, with a `file:line` citation or one-line quote\>.
   > Not acting.

   No PR. No further dispatch. The closing comment trains the next
   session reading this thread and the noise filter.

#### Outcome B — Premise correct + fix is obvious → implement and PR

1. Verify the bot's premise as above.
2. Read the Worker Guidance section below, open a worktree, implement.
3. Open a PR with `Resolves #<this-issue-number>` in the body
   (use THIS issue's number, not the source PR's) so merge auto-closes it.
4. Follow the normal Lifecycle Gate (brief, tests, review-bot-gate,
   merge, postflight).

#### Outcome C — Premise correct but approach is a genuine judgment call

Only use this path if you reach it after Outcomes A and B don't apply:
the bot's finding is real, but the fix requires a decision that is
architectural, policy, breaking-change, or otherwise genuinely outside
what you can resolve autonomously. In that case, post a **decision
comment** with exactly these fields:

- **Premise check:** one line, confirming the finding is real.
- **Analysis:** 2-4 bullets on the trade-offs.
- **Recommended path:** the option you would take if the decision were
  yours, with rationale.
- **Specific question:** the single decision the human needs to make
  (yes/no or pick-one, not open-ended).

Then apply `needs-maintainer-review` and stop. The human wakes up to a
ready-to-approve recommendation, not a blank task.

> **Ambiguity about scope or style is not Outcome C.** Per
> `AGENTS.md` "Reasoning responsibility", the model does the
> thinking and delivers a recommendation. Only escalate what is genuinely
> a maintainer-only decision.

### Worker Guidance

**Files to modify:**

- `.agents/scripts/shared-gh-wrappers-checks.sh:119`

**Implementation steps (Outcome B path):**

1. Read the `diff` block under each inline comment below — it shows the
   exact code the bot was flagging. Open the file only if you need
   surrounding context beyond what the diff tail shows.
2. Read the bot's full comment below the diff — it contains the rationale
   and any suggested change.
3. Verify the premise before implementing (see Outcome A). If the premise
   is wrong, switch to Outcome A instead of burning iterations trying to
   satisfy a wrong suggestion.
4. If multiple comments target the same file, group your edits into one
   logical commit.
5. Run `shellcheck` / `markdownlint-cli2` / project tests as appropriate.

**Verification:**

- Open the new PR with `Resolves #<this-issue>` so this followup is auto-closed on merge.
- If the bot's suggestion was incorrect, close this issue with a Outcome A comment — do not open a no-op PR.

### Inline comments

#### gemini-code-assist on `.agents/scripts/shared-gh-wrappers-checks.sh:119`

[View inline comment](https://github.com/marcusquinn/aidevops/pull/25620#discussion_r3484203640)

```diff
@@ -114,10 +114,12 @@ gh_pr_check_status_rest() {
 	# read-only special variable, which would silently fail under zsh-sourced
 	# interactive use even though the script declares `#!/usr/bin/env bash`.
 	local _check_state=""
+	# shellcheck disable=SC2016 # jq program uses $active as a jq variable.
 	_check_state=$(_gh_checks_api_read "repos/${slug}/commits/${sha}/check-suites" --jq '
-		if (.check_suites | length) == 0 then "none"
-		elif (.check_suites | all(.conclusion == "success" or .conclusion == "skipped" or .conclusion == "neutral")) then "PASS"
-		elif (.check_suites | any(.conclusion == "failure" or .conclusion == "timed_out" or .conclusion == "cancelled")) then "FAIL"
+		(.check_suites | map(select(.conclusion != null or .status != "queued"))) as $active |
```

> ![medium](https://www.gstatic.com/codereviewagent/medium-priority.svg)
> 
> If `.check_suites` is missing or `null` in the API response, `(.check_suites | map(...))` will raise a jq error (`Cannot iterate over null`). To prevent this and ensure robust handling, use the fallback operator `// []` to default to an empty array when `.check_suites` is null or missing.
> 
> ```suggestion
> 		((.check_suites // []) | map(select(.conclusion != null or .status != "queued"))) as $active |
> ```
> 
> <details>
> <summary>References</summary>
> 
> 1. In `jq`, use the fallback operator `//` to provide a default value for expressions that might result in `null`.
> </details>

### PR review summaries

#### gemini-code-assist[bot] review summary

[View review](https://github.com/marcusquinn/aidevops/pull/25620#pullrequestreview-4582846454)

> ## Code Review
> 
> This pull request updates the GitHub check status script to ignore queued check suites with null conclusions when determining the overall PR check status, and adds a corresponding test suite to verify this behavior. The reviewer suggested a robustness improvement in the `jq` filter to handle cases where `.check_suites` is null or missing by using the fallback operator `// []` to prevent potential runtime errors.
> 
> > [!IMPORTANT]
> > The [consumer version of Gemini Code Assist on GitHub](https://developers.google.com/gemini-code-assist/docs/review-repo-code) is being sunset. Starting **June 18, 2026**, new organization installations will be blocked, and all code review activity will officially cease on **July 17, 2026**.
> > For more details on the timeline and next steps, please review the [Help Documentation](https://developers.google.com/gemini-code-assist/docs/deprecations/consumer-code-review).
<!-- aidevops:sig -->
---
[aidevops.sh](https://aidevops.sh) v3.24.2 automated scan.

--- a/.agents/scripts/shared-gh-wrappers-checks.sh
+++ b/.agents/scripts/shared-gh-wrappers-checks.sh
@@ -116,7 +116,7 @@ gh_pr_check_status_rest() {
 	local _check_state=""
 	# shellcheck disable=SC2016 # jq program uses $active as a jq variable.
 	_check_state=$(_gh_checks_api_read "repos/${slug}/commits/${sha}/check-suites" --jq '
-		(.check_suites | map(select(.conclusion != null or .status != "queued"))) as $active |
+		((.check_suites // []) | map(select(.conclusion != null or .status != "queued"))) as $active |
 		if ($active | length) == 0 then "none"
 		elif ($active | all(.conclusion == "success" or .conclusion == "skipped" or .conclusion == "neutral")) then "PASS"
 		elif ($active | any(.conclusion == "failure" or .conclusion == "timed_out" or .conclusion == "cancelled")) then "FAIL"

--- a/.agents/scripts/tests/test-shared-gh-wrappers-checks-null-suites.sh
+++ b/.agents/scripts/tests/test-shared-gh-wrappers-checks-null-suites.sh
@@ -64,6 +64,8 @@ assert_status() {
 }
 
 main() {
+	assert_status "missing check_suites defaults to none" "none" '{}'
+	assert_status "null check_suites defaults to none" "none" '{"check_suites":null}'
 	assert_status "null queued app suites ignored when actions passed" "PASS" '{"check_suites":[{"app":{"slug":"github-actions"},"status":"completed","conclusion":"success"},{"app":{"slug":"coderabbitai"},"status":"queued","conclusion":null}]}'
 	assert_status "all null suites mean no active checks" "none" '{"check_suites":[{"app":{"slug":"coderabbitai"},"status":"queued","conclusion":null}]}'
 	assert_status "terminal failure still fails" "FAIL" '{"check_suites":[{"app":{"slug":"github-actions"},"status":"completed","conclusion":"success"},{"app":{"slug":"github-actions"},"status":"completed","conclusion":"failure"},{"app":{"slug":"coderabbitai"},"status":"queued","conclusion":null}]}'

